### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by AndreaM2429**
Hello there!
I added a new feature to the `BlogApp` project, so I really appreciate you taking the time to review these new implementations, thanks in advance for all your comments and feedback.

# 🎯 Milestones:

📌 Create model classes for tables.
📌 Set up associations between models.
📌 Create custom methods:

- Users
    - A method that returns the 3 most recent posts for a given user.
- Posts
    - A method that updates the posts counter for a user.
    - A method that returns the 5 most recent comments for a given post.
- Comments
    - A method that updates the comments counter for a post.
- Likes
    - A method that updates the likes counter for a post.

## 🎖️ General requirements:
📍 No linter errors
📍 Gitflow
📍 Document work

-------------------------------------------
### Console:

I ran the following commands in the `rails c` and got these results in the console and in the database:
> The counters for each table auto-update when a new register is created.

1. Create users:
```ruby
first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
``` 
image

2. Create post for the first_user:
```ruby
# 4 times
first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
``` 
image
image

3. Create comments for the first_post:
```ruby
# 6 times
Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
``` 
image
image

4. Create likes for the first_post:
```ruby
# 4 times
 Like.create(author: first_user, post: first_post)
``` 
image
image

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
+  has_many :comments, foreign_key: :user_id
+  has_many :likes
+
```
Comment (tresorsawasawa) [inline app/models/user.rb]: - [ ] Good job for adding some of foreign_keys on your different table. However, to correctly set up correctly all the associations according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png), Kindly ensure that you also add the `foreign_key` as in ERD Diagram

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @AndreaM2429 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- Added all the required methods :heavy_check_mark: 
- Added new models :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (AndreaM2429) [reply app/models/user.rb]: Hi @tresorsawasawa, I forgot to add the likes `foreign key` and also I updated the names to be `author_id`. 
Thanks for the feedback.

**Review by fobadara — APPROVED** [review]
Hi @AndreaM2429, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

APPRVED
## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**